### PR TITLE
 Preserve Data in the browser

### PR DESCRIPTION
**Pull request by naji-97**
### **I implemented the following interactions:**

- [ ] When the user changes the content of any input field, the data is saved to the local storage.

- [ ] When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

- [ ] I created a single JavaScript object with all the data from the entire form and save it in local storage.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -129,4 +129,30 @@ const checkLowerCase = (input, event) => {
...
-});
\ No newline at end of file
+});
+
+// Start LocalStorage
+
+const getName = document.querySelector('input[type="text"]');
+const getTextArea = document.querySelector('textarea');
+
+const dataStore = JSON.parse(localStorage.getItem('dataStore')) || [];
```
Comment (amejid) [inline main.js]: - [ ] As it is stated in the [project requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md) we should only store a single javascript object to store the form data.

**Requirement**
Screenshot from 2023-01-07 14-09-18

Every time a form input is updated it is creating an object and stores it in an array of objects. Please check the video below.

https://www.loom.com/share/0e41b7a9725c4264b4566dd4ddf7f122

Kindly update it to use a single object to store the data. ✍️

**Review by amejid — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Correct use of Github flow :heavy_check_mark: 
- Descriptive PR :heavy_check_mark: 

## Required Changes ♻️

- [ ] Please replace the current `linters.yml` file with [this one](https://github.com/microverseinc/linters-config/blob/master/html-css-js/.github/workflows/linters.yml) so the ESLint can run on Github actions.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js [hunk 2]
```diff
@@ -129,4 +129,30 @@ const checkLowerCase = (input, event) => {
...
+
+const showData = (({ name, email, text }) => {
+  getName.value = name;
+  getEmail.value = email;
+  getTextArea.value = text;
+});
+
+dataStore.forEach(showData);
+
+getForm.addEventListener('submit', (e) => {
```
Comment (huseyinbiyik) [inline main.js]: Kindly make sure the data exist after the page load and create one object for it on the local storage as the previous reviewer requested. Tip: Evenlistener needs to listen to the changes on the input, not after the submit. Check [this](https://huseyinbiyik.github.io/) to get inspired.

Currently your project:
Screen Recording 2023-01-07 at 4 40 27 PM

**Review by huseyinbiyik — CHANGES_REQUESTED** [review]
Hi @naji-97 ,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
tenor

 ## Highlights
- Remote collaboration 👌👌


 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @naji-97 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Hub Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes . 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**